### PR TITLE
AXI Agent: Enhancement in the slave section.

### PR DESCRIPTION
**Pull request by AEzzejjari**
Add more maintainability to the memory access method.
Include the ability to support the user signal.
Improve performance by reducing the randomization of all transactions in each clock cycle when the response is not valid.

### CODE REVIEW

File: lib/uvm_agents/uvma_axi5/src/comps/uvma_axi_drv.sv
Comment (khandelwaltanuj) [inline lib/uvm_agents/uvma_axi5/src/comps/uvma_axi_drv.sv:1]: Hi @AEzzejjari 

I am a little confused by these changes. I believe the randomization of m_data/m_last should be dependent of m_len. 
Thats what the "m_txn.randomize" does automatically. 
I am wondering does it work in the case where a reponse with multiple flit is expected ? 
May be I am missing something here ? 

Thanks and Regards
Tanuj

Comment (khandelwaltanuj) [reply lib/uvm_agents/uvma_axi5/src/comps/uvma_axi_drv.sv:1]: Ah I get it, it is the case when r_txn is null and we are just driving random stuff ?

Comment (AEzzejjari) [reply lib/uvm_agents/uvma_axi5/src/comps/uvma_axi_drv.sv:1]: Hello @khandelwaltanuj 

This randomization is made to accelerate performance because m_trs.randomize consumes a lot of time and is applied every cycle.

Thank you
Alae Eddine

**Comment by JeanRochCoulon** [comment]
@khandelwaltanuj do you approve the modification ?   I wait for your GO to merge.

**Comment by khandelwaltanuj** [comment]
Hi @JeanRochCoulon 

A visual review seems to be correct. But I have a question before I approve the changes: Is there a way I can use your fork and compile the code ? I have created a PR to add little example, which I can everutally use to see if the changes of AXI are compiling with correctly with VCS and QUESTA ? 

thanks and regards
Tanuj

**Comment by AEzzejjari** [comment]
Hello @khandelwaltanuj

You can get this PR locally and compile it in your environment. 
Here's a link to show you how : https://docs.github.com/fr/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/checking-out-pull-requests-locally

Regards
Alae Eddine

**Comment by khandelwaltanuj** [comment]
Hi @AEzzejjari 

Thanks a lot.

I tried different things but its doesnt seems to work.

[tk223379@nc036 core-v-verif]$ git fetch origin pull/2558/head:AXI_enh
fatal: Couldn't find remote ref pull/2558/head
[tk223379@nc036 core-v-verif]$ git fetch origin pull/2558/ThalesSiliconSecurity:AXI_enh
fatal: Couldn't find remote ref pull/2558/ThalesSiliconSecurity

For this PR, I will approve it now. Thoug I would not mind, if you can help me get the PR locally. 

Regards
Tanuj

**Review by khandelwaltanuj — APPROVED** [review]

**Comment by MikeOpenHWGroup** [comment]
Hi @khandelwaltanuj, I am not sure about those `git fetch` commands.  Try this:
```
git clone -b AXI_enh git@github.com:ThalesSiliconSecurity/core-v-verif.git
```
This will clone the head of the repo/branch that is the source of this PR.

**Pull request merged**
